server/fixblobs.go

```diff
@@ -24,3 +24,35 @@ func fixBlobs(dir string) error {
 		return nil
 	})
 }
+
+// fixManifests walks the provided dir and replaces (":") to ("%") for all
+// manifest files on non-Windows systems.
+func fixManifests(dir string) error {
+	return filepath.Walk(dir, func(oldPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
```

Comment:
Why not rename just the directory?

Reply:
It was easier to reason about how to find the host once I had a file (e.g. tag).

Reply:
I believe we should only need to replace `%` for the dirs in `~/.ollama/models/manifests/`, not sure if the code below this is required?

Reply:
That should only change the host part of the path. @mxyng @jmorganca Do you have any suggestions for making it more clear? It's my understanding this is a migration script only, not a critical path, so I gave it a quick pass to get us form a to b.

This patch comes with tests. Do you have additional test cases in mind that prove this renames more than the host part of the path?

Reply:
I think what Jeff is saying is that we can simplify this a bit to only rename the manifest directories, and not worry about their contents. The ports should only exist on the manifest directories since we don't allow them within the model names/tags.

<img width="366" alt="image" src="https://github.com/ollama/ollama/assets/5853428/00b02c6e-df6f-45c3-b7be-d9639fec2274">

More concretely we wont need this test:
```
		{path: []string{"x:y/h:p/n/m/t"}, want: []string{"x:y/h%p/n/m/t"}},
```